src/encoded/static/components/motifs.js

```diff
@@ -285,6 +352,60 @@ export const Motifs = (props) => {
         alignedEndCoordinate = Math.max(p.end, alignedEndCoordinate);
     });
 
+    // trimming the reference sequence to widest window of pwms
+    const referenceLength = alignedEndCoordinate - alignedStartCoordinate;
+    const referenceStart = alignedStartCoordinate - props.context.sequence.start;
+    const referenceSequence = props.context.sequence.sequence.slice(referenceStart, referenceStart + referenceLength);
+
+    const trackScrolling = () => {
+        if (tableTop && tableTop.current && refContainer && refContainer.current && refPlaceholder && refPlaceholder.current) {
+            if (tableTop.current.getBoundingClientRect().top <= 27) {
+                refContainer.current.classList.add('fixed-position');
+                refPlaceholder.current.setAttribute('style', `height:${refContainer.current.getBoundingClientRect().height}px`);
+            } else {
+                refContainer.current.classList.remove('fixed-position');
+                refPlaceholder.current.setAttribute('style', 'width:0px');
+            }
+        }
+    };
+
+    React.useEffect(() => {
+        if (limit === 0) {
+            document.addEventListener('scroll', trackScrolling);
+        }
+
+        // drawing reference sequence as a pwm so it lines up with the others
+        require.ensure(['d3', 'd3-sequence-logo'], (require) => {
+            setD3Lib(require('d3'));
+            const sequenceLogos = logos; // This is for local development when changes are needed to d3-sequence-logo.
+            // this.sequenceLogos = require('d3-sequence-logo');
+
+            const refSeq = referenceSequence.split('');
+            const fakePWM = [];
+
+            refSeq.forEach((nucleotide) => {
+                const maxHeight = 1000;
+                if (nucleotide === 'A') {
+                    fakePWM.push([maxHeight, 0, 0, 0]);
+                } else if (nucleotide === 'C') {
+                    fakePWM.push([0, maxHeight, 0, 0]);
+                } else if (nucleotide === 'G') {
+                    fakePWM.push([0, 0, maxHeight, 0]);
+                } else {
+                    fakePWM.push([0, 0, 0, maxHeight]);
+                }
+            });
+
+            if (d3lib) {
+                sequenceLogos.entryPoint(refToReference.current, fakePWM, d3lib, alignmentCoordinate, alignedStartCoordinate, referenceStart, '+', true);
+            }
+
+            return () => {
+                document.removeEventListener("scroll", trackScrolling);
+            }
+        });
+    }, [d3lib, props]);
```

Comment:
I see now. `props` is an object owned by React so that gets tricky in dependency arrays. But you can use the properties you use _inside_ of `props`. So `limit`, `referenceSequence`, `refToReference.current`, `alignmentCoordinate`, `alignedStartCoordinate`, `referenceStart`, `trackScrolling` (a function, but it gets reallocated on every render; you could make it unchanging by wrapping that function in `useCallback()`). ESLint might want `setD3Lib` as well, though it has no chance of changing since it’s a state-setting function.

Reply:
like just directly? Or do I go: `props.limit` etc

Reply:
In this case, I think it’s just `props.limit`. Every other dependency is a local variable. That’s one reason I destructure in the function declaration so that I don’t have `props` at all.